src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1194,17 +1194,17 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
                 }
             }
 
-            // TODO allow customization of this.
-            // Chrome is using h3-29 for protocol ID as of 12/2020. This is likely to be the alt-svc
-            // value until HTTP/3 is finalized.
-            // More info: https://blog.chromium.org/2020/10/chrome-is-deploying-http3-and-ietf-quic.html
             if (ServerOptions.EnableAltSvc && _httpVersion < Http.HttpVersion.Http3)
             {
+                // TODO: Perf. Avoid allocating enumerator and property's LINQ.
+                // https://github.com/dotnet/aspnetcore/issues/34468
                 foreach (var option in ServerOptions.ListenOptions)
                 {
                     if ((option.Protocols & HttpProtocols.Http3) == HttpProtocols.Http3)
                     {
-                        responseHeaders.HeaderAltSvc = $"h3-29=\":{option.IPEndPoint!.Port}\"; ma=84600";
+                        // TODO: Perf. Create string once instead of per-request.
+                        // https://github.com/dotnet/aspnetcore/issues/34468
+                        responseHeaders.HeaderAltSvc = $"h3=\":{option.IPEndPoint!.Port}\"; ma=84600";
```

Comment:
Latest draft has finalized "h3"

Reply:
I was testing the ALPN code with similar values. I think we should include both h3 and h3-29 for compat for at least a milestone, that's really common on websites I've checked. h3 alpn didn't work with HttpClient yet.

Reply:
I found the opposite with HttpClient: h3-29 doesn’t work and h3 does. That’s why I changed it in this PR.

Reply:
Chrome (and Edge) support "h3"

![image](https://user-images.githubusercontent.com/303201/126086373-7f79a0d4-c697-47b1-affd-b187afc4449f.png)

If HttpClient and major browsers support h3 then I think it's fine to completely switch in RC1.